src/mantarray_desktop_app/file_writer.py

```diff
@@ -1006,6 +1032,162 @@ def _handle_performance_logging(self) -> None:
+
+        if not os.path.exists(customer_failed_uploads_dir):
+            os.makedirs(customer_failed_uploads_dir)
+
+        file_name = f"{sub_dir}.zip"
+        zipped_file = os.path.join(zipped_recordings_dir, customer_account_id, file_name)
+        updated_zipped_file = os.path.join(failed_uploads_dir, customer_account_id, file_name)
+
+        # store failed zip file in failed uploads directory to check at next startup
+        if os.path.exists(zipped_file):
+            shutil.move(zipped_file, updated_zipped_file)
+        # Lucy (10/22/2021): this would be where to notify FE of error to display
+
+    def _process_failed_upload_files_on_setup(self) -> None:
+        """Re-upload any failed files on setup_before_loop.
+
+        Will try to process all uploads in failed_uploads directory.
+        Will redirect processed .zip file after successful upload to
+        zipped_recordings directory, but will not move on error.
+        """
+        if self._stored_customer_settings is not None:
+            # this path is created in electrons main process, but this check is for testing purposes
+            failed_uploads_dir = self._stored_customer_settings["failed_uploads_dir"]
+            zipped_recordings_dir = self._stored_customer_settings["zipped_recordings_dir"]
+            stored_customer_ids = self._stored_customer_settings["stored_customer_ids"]
+
+            if os.path.exists(failed_uploads_dir):  # for testing
+                for customer_dir in os.listdir(failed_uploads_dir):
+                    customer_passkey = stored_customer_ids[customer_dir]
+                    customer_failed_uploads_dir = os.path.join(failed_uploads_dir, customer_dir)
+
+                    for file_name in os.listdir(customer_failed_uploads_dir):
+                        upload_thread = ErrorCatchingThread(
+                            target=uploader,
+                            args=(
+                                customer_failed_uploads_dir,
+                                file_name,
+                                zipped_recordings_dir,
+                                customer_dir,
+                                customer_passkey,
+                            ),
+                        )
+                        upload_thread.start()
+                        thread_dict = {
+                            "failed_upload": True,
+                            "customer_account_id": customer_dir,
+                            "thread": upload_thread,
+                            "auto_delete": False,
+                            "file_name": file_name,
+                        }
+                        self._upload_threads_container.append(thread_dict)
+                        # Lucy (10/22/2021): figure out how to handle if delete local files had been selected on original customer settings and how to handle the zip file
+
+    def _check_upload_statuses(self) -> None:
```

Comment:
This method is a great one for using the early exit pattern I mentioned in the other comment. `return`ing after line 1157 and `continue`ing after line 1165 would save two indents from the inner most branch of the loop. But of course it comes down to preference, both are fine